src/utils/prompt-for-missing-options.ts

```diff
@@ -17,35 +17,32 @@ export async function promptForMissingOptions(
   options: RawOptions,
   solidityFrameworkChoices: SolidityFrameworkChoices,
 ): Promise<Options> {
-  const cliAnswers = Object.fromEntries(Object.entries(options).filter(([, value]) => value !== null));
-  const questions = [
-    {
-      type: "input",
-      name: "project",
+  const project =
+    options.project ??
+    (await input({
       message: "Your project name:",
-      default: defaultOptions.project,
+      default: defaultOptions.project as string,
```

Comment:
Maybe for `defaultOptions` defined we do it like this: 

```ts
// default values for unspecified args
const defaultOptions = {
  project: "my-dapp-example",
  solidityFramework: null,
  install: true,
  dev: false,
  externalExtension: null,
  help: false,
} as const satisfies RawOptions;
```

So that we don't have to assert things here